src/audio/module_adapter/module/volume/volume.c

```diff
@@ -1135,7 +1135,7 @@ static vol_zc_func vol_get_zc_function(struct comp_dev *dev,
 
 	/* map the zc function to frame format */
 	for (i = 0; i < ARRAY_SIZE(zc_func_map); i++) {
-		if (sinkb->stream.frame_fmt == zc_func_map[i].frame_fmt)
+		if (sinkb->stream.valid_sample_fmt == zc_func_map[i].frame_fmt)
```

Comment:
The name of "valid_sample_fmt" is a bit strange (in "struct audio_stream") but it has the same type "enum sof_ipc_frame", so this seems good.